service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RaiseQueryController.java

```diff
@@ -0,0 +1,153 @@
+import uk.gov.hmcts.reform.fpl.enums.UserRole;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.service.UserService;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+@Slf4j
+@RestController
+@RequestMapping("/callback/raise-query")
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class RaiseQueryController extends CallbackController {
+
+    private final UserService userService;
+    private static final Map<CaseRole, String> COLLECTION_MAPPING = initialiseCollectionMapping();
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        logAllQueryCollections(caseDetails);
+
+        if (userService.isCafcassUser()) {
+            String cafcassCollection = "qmCaseQueriesCollectionCafcass";
+
+            log.info("Current logged-in user's case role is {}", UserRole.CAFCASS);
+            log.info("Query collection for user role {} is {}.", UserRole.CAFCASS, cafcassCollection);
+            caseDetails.getData().putIfAbsent(cafcassCollection, null);
+            log.info("Setting {} to value {}", cafcassCollection, caseDetails.getData().get(cafcassCollection));
+        } else {
+            Set<CaseRole> currentUserRoles = userService.getCaseRoles(caseData.getId());
+            log.info("Current logged-in user's case roles are: {}", currentUserRoles);
+            initialiseRelevantQueryCollectionsForUser(caseDetails, currentUserRoles);
+        }
+
+        log.info("Final values of query collections: ");
+        logAllQueryCollections(caseDetails);
+
+        return respond(caseDetails);
+    }
+
+    private void initialiseRelevantQueryCollectionsForUser(CaseDetails caseDetails, Set<CaseRole> currentUserRoles) {
+        CaseRole userQueryCollectionRole = currentUserRoles.stream()
+            .filter(COLLECTION_MAPPING::containsKey)
+            .findFirst()
+            .orElse(null);
+
+        if (userQueryCollectionRole != null) {
+            String queryCollectionKey = COLLECTION_MAPPING.get(userQueryCollectionRole);
+            log.info("Query collection for user role {} is {}.", userQueryCollectionRole, queryCollectionKey);
+            caseDetails.getData().putIfAbsent(queryCollectionKey, null);
+            log.info("Setting {} to value {}", queryCollectionKey, caseDetails.getData().get(queryCollectionKey));
+        }
+    }
+
+    private void logQueryCollection(CaseDetails caseDetails, String collectionKey) {
+        if (caseDetails.getData().containsKey(collectionKey)) {
+            log.info("{} is present with value: {}", collectionKey, caseDetails.getData().get(collectionKey));
+        } else {
+            log.info("{} is not present on case data.", collectionKey);
+        }
```

Comment:
logging will need to be removed before deployment, as field contents may contain PII.